nrf_modem/doc/at-interface.rst

```diff
@@ -0,0 +1,197 @@
+
+Sending AT commands
+*******************
+
+The application can use :c:func:`nrf_modem_at_printf` to send a formatted AT command to the modem when it is interested only in the generic result of the operation, e.g. "OK", "ERROR", "+CME ERROR" or "+CMS ERROR".
+This function works similarly to :c:func:`printf` from the standard C library, and it supports all the format specifiers supported by the :c:func:`printf` implementation of the selected C library.
+
+The following snippet shows how to use :c:func:`nrf_modem_at_printf` to send a formatted AT command to the modem and check the result of the operation.
+
+.. code-block:: c
+
+	int cfun_control(int mode)
+	{
+		int err;
+
+		err = nrf_modem_at_printf("AT+CFUN=%d", mode);
+		if (err = 0) {
+			/* OK, success */
+		} else if (err < 0) {
+			/* Failed to send command, err is an nrf_errno */
+		} else if (err > 0) {
+			/* Command was sent, but response is not "OK" */
+			switch(nrf_modem_at_err_type(err)) {
+			case NRF_MODEM_AT_ERROR:
+				/* Modem returned "ERROR" */
+				printf("error");
+				break;
+			case NRF_MODEM_AT_CME_ERROR:
+				/* Modem returned "+CME ERROR" */
+				printf("cme error: %d", nrf_modem_at_err(err));
+				break;
+			case NRF_MODEM_AT_CMS_ERROR:
+				/* Modem returned "+CMS ERROR" */
+				printf("cms error: %d", nrf_modem_at_err(err));
+				break;
+			}
+		}
+		return err;
+	}
+
+	int foo(void)
+	{
+		/* Send AT+CFUN=1 */
+		cfun_control(1);
+		/* Send AT+CFUN=4 */
+		cfun_control(4);
+	}
+
+Any return value other than zero indicates an error.
+Negative values mean that the modem library has failed to send the AT command, and they represent an nrf_errno that indicates the reason for the failure.
+Positive values mean that the modem has received the AT command and responded with an error. When a positive value is returned, the error type can be retrieved using the :c:func:`nrf_modem_at_err_type` helper function, and the error value (in case of CME or CMS errors) can be retrieved with the :c:func:`nrf_modem_at_err` helper function.
+
+Reading data from an AT response
+********************************
+
+Use :c:func:`nrf_modem_at_scanf` to send an AT command to the modem and parse the response according to a specified format.
+This function works similarly to :c:func:`scanf` from the standard C library, and it supports all the format specifiers supported by the :c:func:`scanf` implementation of the selected C library.
+
+The following snippet shows how to use :c:func:`nrf_modem_at_scanf` to read the modem network registration status via "AT+CEREG?".
+
+.. code-block:: c
+
+	void cereg_read(void)
+	{
+		int rc;
+		int status;
+
+		/* The `*` sub-specifier discards the result of the match.
+		 * The data is read but it is not stored in any argument.
+		 */
+		rc = nrf_modem_at_scanf("AT+CEREG?", "+CEREG: %*d,%d", &status);
+
+		/* Upon returning, `rc` contains the number of matches */
+		if (rc > 1) {
```

Comment:
It may be possible that a response may contain more parameters than what we are interested in, in which case it's best to make sure that the number of matches are _at least_ the number we expect. I am not 100% sure for responses, but for notifications the number of parameters is dependent on the level that the notification is enabled at.

Reply:
But isn't the return value related to the number of arguments we've provided for a call, not the actual number of parameters in the parsed string? So that the function cannot return more matches then the number of arguments we've actually provided? That's why I think it'd make more sense to check the exact number of matches to confirm that we've filled all of the variables we wanted.

Reply:
Ok, will update